process/transaction/shardProcess.go

```diff
@@ -1018,15 +1115,22 @@ func (txProc *txProcessor) executeFailedRelayedUserTx(
 	}
 
 	totalFee := txProc.economicsFee.ComputeFeeForProcessing(userTx, userTx.GasLimit)
+	moveBalanceGasLimit := txProc.economicsFee.ComputeGasLimit(userTx)
+	gasToUse := userTx.GetGasLimit() - moveBalanceGasLimit
+	processingUserFee := txProc.economicsFee.ComputeFeeForProcessing(userTx, gasToUse)
 	if txProc.enableEpochsHandler.IsFlagEnabled(common.FixRelayedMoveBalanceFlag) {
-		totalFee = txProc.economicsFee.ComputeTxFee(userTx)
+		moveBalanceUserFee := txProc.economicsFee.ComputeMoveBalanceFee(userTx)
```

Comment:
can't you reuse the common code for this?

Reply:
some of the above variables are used in further ifs